crates/markdown/src/parser.rs

```diff
@@ -123,6 +123,28 @@ pub fn parse_links_only(text: &str) -> Vec<(Range<usize>, MarkdownEvent)> {
     events
 }
 
+pub fn safe_markdown_slice(source: &str, range: std::ops::Range<usize>) -> (&str, usize) {
```

Comment:
Wouldn't it make sense for this to return `Option<(&str, usize)>`? That way you don't have to return `("", ...)`? Or am I missing something and the `""` is useful?

Reply:
You're right. Updated the PR with `Option<(&str, usize)>` return type.